tests/data_engine/test_querying.py

```diff
@@ -27,6 +27,25 @@ def test_simple_filter(ds):
     assert q.to_dict() == expected
 
 
+def test_versioning(ds):
+    add_int_fields(ds, "x")
+    add_int_fields(ds, "y")
+
+    ds2 = ((ds[ds[Field("x", as_of_timestamp=123.99)] > 1]) &
+           (ds[ds[Field("x", as_of_timestamp=345)] > 2]) |
+           (ds[ds[Field("y", as_of_timestamp=789)] > 3])).\
```

Comment:
If you want to have this be an example of a "complex" query for versioning, I'd switch the second X for a non-timestamped just for giggles, but that's up to you